lua/orgmode/files/headline.lua

```diff
@@ -288,6 +288,28 @@ function Headline:set_tags(tags)
   vim.api.nvim_buf_set_text(bufnr, pred_end_row, pred_end_col, pred_end_row, end_col, { text })
 end
 
+---@param tag string
+---@param onoff? boolean true=>add, false=>remove, nil=>toggle the tag
```

Comment:
Instead of providing the 2nd argument, let's split this into several methods, if necessary:

1. `toggle_tag` - same as if this value is `nil`.
2. `add_tag` - same as if this value is `true`
3. `remove_tag` - same as if this value is `false`

It will introduce some duplication but the usage will be more understandable when reading.

Reply:
That's entirely fair! I've added the new methods and some tests so that all cases are covered for each of them :+1: